Issue25640 tob ci

Malarg opened the pull request:
resolves #25640
tour of beam welcome page integration tests

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[![Build python source distribution and wheels](https://github.com/apache/beam/workflows/Build%20python%20source%20distribution%20and%20wheels/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[![Python tests](https://github.com/apache/beam/workflows/Python%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[![Java tests](https://github.com/apache/beam/workflows/Java%20Tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[![Go tests](https://github.com/apache/beam/workflows/Go%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

alexeyinkin committed:
Integration test to load the default example of the default SDK and change the example (#24730) (#24729)

alexeyinkin committed:
Fix formatting and README (#24730)

alexeyinkin committed:
Support collection v1.17.0 (#24730)

alexeyinkin committed:
LoadingIndicator on chaning examples, remove duplicating licenses (#24730)

alexeyinkin committed:
Add a missing license header (#24730)

alexeyinkin committed:
Merge branch 'master' into issue24730_integration_changing-example

alexeyinkin committed:
Integration test for changing SDK and running code (#24779) (#382)

* Integration test for changing SDK and running code (#24779)

* Rename an integration test (#24779)

* Use enum to switch SDK in integration test (#24779)

* Find SDK in a dropdown by key (#24779)

* Add a TODO (#24779)

* Fix exports (#24779)

alexeyinkin committed:
Merge branch 'master' into issue24730_integration_changing-example

alexeyinkin committed:
Issue24779 integration changing sdk from 24370 (#387)

* Integration test for changing SDK and running code (#24779)

* Rename an integration test (#24779)

* Use enum to switch SDK in integration test (#24779)

* Find SDK in a dropdown by key (#24779)

* Add a TODO (#24779)

* Fix exports (#24779)

* Integration tests miscellaneous UI (#383)

* miscellaneous ui integration tests

* reverted pubspec.lock

* gradle tasks ordered alhpabetically

* integration tests refactoring

* clean code

* integration tests miscellaneous ui fix pr

* rename method

* added layout adaptivity

* A minor cleanup (#24779)

Co-authored-by: Dmitry Repin <mr.malarg@gmail.com>

Malarg committed:
integration tests run and editing

Malarg committed:
Merge branch 'master' into playground_integration_test_run_and_editing

Malarg committed:
example selector test

Malarg committed:
minor fixes

Malarg committed:
rat

Malarg committed:
fix pr

Malarg committed:
minor

Malarg committed:
Merge branch 'master' into playground_integration_test_run_and_editing

Malarg committed:
minor

Malarg committed:
rat

Malarg committed:
integration test finder written

Malarg committed:
Merge branch 'master' into playground_integration_test_run_and_editing

Malarg committed:
integration test minor fixes

Malarg committed:
minor fixes

Malarg committed:
removed comment

Malarg committed:
minor fixes

Malarg committed:
Merge branch 'master' into playground_integration_test_run_and_editing

Malarg committed:
playground integration tests minor fixes

Malarg committed:
integration test pumpAnSettleNoException

Malarg committed:
integration test shortcut refactor

Malarg committed:
integration test another changing shortcuts running

Malarg committed:
upgrade to flutter 3.7.1

Malarg committed:
workaround comment

Malarg committed:
playground frontend updated major versions

Malarg committed:
issues 25329 25331 25336

Malarg committed:
Merge branch 'master' into issue25329_fix-cors-issue-when-no-internet

Malarg committed:
25329 extract connectivity extension to separate file

alexeyinkin committed:
Merge remote-tracking branch 'apache/master' into playground_integration_test_run_and_editing

# Conflicts:
#	playground/frontend/assets/translations/en.yaml

alexeyinkin committed:
Upgrade Flutter to 3.7.3 in integration tests (#24730)

alexeyinkin committed:
Fix integration test (#24730)

alexeyinkin committed:
Merge remote-tracking branch 'origin/issue25337_flutter371' into playground_integration_test_run_and_editing

# Conflicts:
#	playground/frontend/playground_components/pubspec.yaml
#	playground/frontend/pubspec.yaml

Malarg committed:
fix cors issue and added mouse scroll to tags

alexeyinkin committed:
Upgrade Flutter in Dockerfile (#24720)

Malarg committed:
sorting moved to model

Malarg committed:
sorting moved to model

Malarg committed:
sorting moved to model

Malarg committed:
Merge branch 'master' into playground_integration_test_run_and_editing

Malarg committed:
bugs fix

Malarg committed:
issue 25278

Malarg committed:
fix pr

Malarg committed:
quites fix in en.yaml

alexeyinkin committed:
Fix not loading default example (#25528)

Malarg committed:
Merge branch 'master' into issue25329

Malarg committed:
Merge branch 'playground_integration_test_run_and_editing' into issue25329_fix-cors-issue-when

Malarg committed:
Merge branch 'master' into playground_integration_test_run_and_editing

Malarg committed:
Merge branch 'playground_integration_test_run_and_editing' into issue25329_fix-cors-issue-when-no-internet

Malarg committed:
fix compile error

alexeyinkin committed:
Refactor output tabs, test embedded playground (#25136) (#439)

* Refactor output tabs, test embedded playground (#25136)

* Clean up (#25136)

alexeyinkin committed:
Change example paths to IDs in integration tests

Malarg committed:
issue25640 tob ci

Malarg committed:
fix tob ci

Malarg committed:
rename ci process

Malarg committed:
test add new line to main

Malarg committed:
test add new line to main

Malarg committed:
commented unit test run

Malarg committed:
issue25640 changed server path

Malarg committed:
Merge branch 'playground_integration_test_run_and_editing' into issue25640_tob-ci

Malarg committed:
issue25640 tests on welcome page

Malarg committed:
Merge branch 'master' into playground_integration_test_run_and_editing

Malarg committed:
Merge branch 'master' into issue25640_tob-ci

Malarg committed:
Merge branch 'playground_integration_test_run_and_editing' into issue25640_tob-ci

Malarg committed:
deleted config.g.dart

alexeyinkin commented on .github/workflows/tour_of_beam_frontend_test.yml:
```diff
@@ -0,0 +1,86 @@
+
+on:
+  push:
+    paths: ['learning/tour-of-beam/frontend/**']
+    branches: ['**']
+  pull_request:
+    paths: ['learning/tour-of-beam/frontend/**']
+    branches: ['**']
```
Also `playground_components`.

Malarg replied:
added

alexeyinkin commented on learning/tour-of-beam/frontend/integration_test/common/common_finders.dart:
```diff
@@ -17,27 +17,19 @@
-void main() {
-  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+extension CommonFindersExtension on CommonFinders {
+  Finder sdkDropdwn() {
```
```suggestion
  Finder sdkDropdown() {
```

Malarg replied:
fixed

alexeyinkin commented on learning/tour-of-beam/frontend/integration_test/welcome_page_test.dart:
```diff
@@ -0,0 +1,118 @@
+      findsOneWidget,
+    );
+
+    expect(
+      find.dropdownMenuItemWithText(sdk.title),
+      findsNothing,
+    );
```
Check that there is no SDK dropdown if the SDK is not selected.

alexeyinkin commented on learning/tour-of-beam/frontend/integration_test/welcome_page_test.dart:
```diff
@@ -0,0 +1,118 @@
+  await wt.tapAndSettle(find.sdkDropdwn());
+
+  for (final sdk in sdks) {
+    expect(
+      find.dropdownMenuItemWithText(sdk.title),
+      findsNWidgets(2), //DropdownButton widget peculiarity
+    );
```
Is there a way to check that there is only one visible widget with the given text?

alexeyinkin commented on learning/tour-of-beam/frontend/integration_test/welcome_page_test.dart:
```diff
@@ -0,0 +1,118 @@
+  checkModulesDisplayed();
+}
+
+void checkModulesDisplayed() {
```
```suggestion
void _checkModulesDisplayed() {
```

Malarg replied:
fixed

alexeyinkin commented on learning/tour-of-beam/frontend/lib/config.dart:
```diff
@@ -28,12 +28,12 @@ const cloudFunctionsBaseUrl = 'https://'
 const String kAnalyticsUA = 'UA-73650088-2';
 const String kApiClientURL =
-    'https://backend-router-beta-dot-apache-beam-testing.appspot.com';
+    'https://router.playground-uat.xyz';
```
```suggestion
    'https://router.play-dev.beam.apache.org';
```

Malarg replied:
fixed

alexeyinkin commented on learning/tour-of-beam/frontend/settings.gradle:
```diff
@@ -0,0 +1,19 @@
+ * limitations under the License.
+ */
+
+rootProject.name = 'apache-beam-tob-frontend'
```
If the project is called `:learning:tour-of-beam:frontend` (the default), this file is not required.

Malarg replied:
deleted

alexeyinkin commented on playground/frontend/integration_test/standalone_miscellaneous_ui_test.dart:
```diff
@@ -18,14 +18,14 @@
 
 import 'package:flutter_test/flutter_test.dart';
 import 'package:integration_test/integration_test.dart';
+import 'package:playground_components_dev/src/common_tests/toggle_brightness_mode_test.dart';
```
Do not import from `src`, export it.

Malarg replied:
done

Malarg committed:
issue25640 pr fixes

alexeyinkin commented on .github/workflows/tour_of_beam_frontend_test.yml:
```diff
@@ -0,0 +1,86 @@
+
+      - name: 'Install Dependencies'
+        run: |
+          cd playground/frontend/playground_components && flutter pub get && cd ..
+          cd playground_components_dev && flutter pub get && cd ..
+          flutter pub get && cd ../../
+          cd learning/tour-of-beam/frontend && flutter pub get
```
```suggestion
          cd playground/frontend/playground_components && flutter pub get && cd -
          cd playground/frontend/playground_components_dev && flutter pub get && cd -
          cd learning/tour-of-beam/frontend && flutter pub get && cd -
```

alexeyinkin commented on learning/tour-of-beam/frontend/integration_test/welcome_page_test.dart:
```diff
@@ -0,0 +1,118 @@
+}
+
+Future<void> _checkSdksLoadedCorrectly(WidgetTester wt) async {
+  //The source of truth of loaded sdks
```
```suggestion
  // The source of truth of loaded sdks.
```

Malarg committed:
Update .github/workflows/tour_of_beam_frontend_test.yml

Co-authored-by: alexeyinkin <leha@inkin.ru>

Malarg committed:
Update learning/tour-of-beam/frontend/integration_test/welcome_page_test.dart

Co-authored-by: alexeyinkin <leha@inkin.ru>

alexeyinkin committed:
Improve tests (#25640)

robertwb committed:
Use the cord ctype for proto generation, where available.

This is natural as this field is often large and composed of
the concatenation of many small elements, which can lead to
significant performance improvements.

Malarg committed:
issue25640 tour page tests

Malarg committed:
Merge branch 'issue25640_tob-ci' of https://github.com/akvelon/beam into issue25640_tob-ci

Malarg commented:
@alexeyinkin review please

liferoad committed:
Add one example to learn beam by doing (#25719)

* Add one example to learn beam by doing

* add license

* clear the output

* Polished the notebook based on the comments

* cleared out the output

* Changed the answer views

* Update some cells

* Polished the words

* Update examples/notebooks/get-started/learn_beam_basics_by_doing.ipynb

Co-authored-by: tvalentyn <tvalentyn@users.noreply.github.com>

---------

Co-authored-by: xqhu <xqhu@google.com>
Co-authored-by: tvalentyn <tvalentyn@users.noreply.github.com>

lostluck committed:
Ensure truncate element is wrapped in *FullValue (#25908)

Co-authored-by: lostluck <13907733+lostluck@users.noreply.github.com>

nickuncaged1201 committed:
Pubsub test client fixup (#25907)

* make incoming message public

* add set clock method to test client

* spotless

Malarg commented on learning/tour-of-beam/frontend/integration_test/tour_page_test.dart:
```diff
@@ -0,0 +1,222 @@
+  expectSimilar(startHandlePosition.dx, movedHandlePosition.dx - 100);
+}
+
+Future<void> _checkRunCodeWorks(WidgetTester wt) async {
```
No, I don't think so. It's very similar to `runCustomJava` method in playground tests, but I think that using extracted method will be weird, because all that we can extract - put code, that prints given string to code field. For clear understanding I think it's better to keep this logic without extracting for a while. Possible we will extract this, when cases will be more than 2.

Malarg committed:
pr fix

liferoad committed:
Raise the Runtime error when DoFn.process uses both yield and return (#25743)

Co-authored-by: xqhu <xqhu@google.com>

robertwb committed:
Merge pull request #25802 Use the cord ctype for proto generation, where available.

jackdingilian committed:
Advance DetectNewPartition's watermark by aggregating the (#25906)

watermark of ReadChangeStreamPartitions

reuvenlax committed:
Merge pull request #25723: #25722 Add option to propagate successful storage-api writes

Malarg committed:
Merge branch 'master' into issue25640_tob-ci

Malarg committed:
removed import

Malarg committed:
pr fix

Malarg committed:
fix test
